airbyte-integrations/connectors/source-asana/source_asana/manifest.yaml

```diff
@@ -0,0 +1,2595 @@
+    retriever:
+      $ref: "#/definitions/retriever"
+      requester:
+        $ref: "#/definitions/requester"
+        use_cache: true
+    $parameters:
+      path: "/workspaces"
+
+  workspaces_partition_router:
+    type: SubstreamPartitionRouter
+    parent_stream_configs:
+      - stream: "#/definitions/workspaces_stream"
+        parent_key: "gid"
+        partition_field: "workspace_gid"
+
+  tags_stream:
+    $ref: "#/definitions/base_stream"
+    name: "tags"
+    schema_loader:
+      type: InlineSchemaLoader
+      schema:
+        $ref: "#/definitions/tags_schema"
+    $parameters:
+      path: "/workspaces/{{ stream_slice['workspace_gid'] }}/tags"
+    retriever:
+      $ref: "#/definitions/retriever"
+      partition_router:
+        $ref: "#/definitions/workspaces_partition_router"
+
+  users_stream:
+    $ref: "#/definitions/base_stream"
+    name: "users"
+    schema_loader:
+      type: InlineSchemaLoader
+      schema:
+        $ref: "#/definitions/users_schema"
+    $parameters:
+      path: "/users"
+
+  users_partition_router:
+    type: SubstreamPartitionRouter
+    parent_stream_configs:
+      - stream: "#/definitions/users_stream"
+        parent_key: "gid"
+        partition_field: "users_gid"
+
+  teams_stream:
+    $ref: "#/definitions/base_stream"
+    name: "teams"
+    schema_loader:
+      type: InlineSchemaLoader
+      schema:
+        $ref: "#/definitions/teams_schema"
+    $parameters:
+      path: "/organizations/{{ stream_slice['workspace_gid'] }}/teams"
+    retriever:
+      $ref: "#/definitions/retriever"
+      partition_router:
+        $ref: "#/definitions/workspaces_partition_router"
+
+  teams_partition_router:
+    type: SubstreamPartitionRouter
+    parent_stream_configs:
+      - stream: "#/definitions/teams_stream"
+        parent_key: "gid"
+        partition_field: "team_gid"
+
+  team_memberships_stream:
+    $ref: "#/definitions/base_stream"
+    name: "team_memberships"
+    schema_loader:
+      type: InlineSchemaLoader
+      schema:
+        $ref: "#/definitions/team_memberships_schema"
+    $parameters:
+      path: "/teams/{{ stream_slice['team_gid'] }}/team_memberships"
+    retriever:
+      $ref: "#/definitions/retriever"
+      paginator:
+        $ref: "#/definitions/paginator_base"
+      partition_router:
+        $ref: "#/definitions/teams_partition_router"
+
+  attachments_compact_stream:
+    $ref: "#/definitions/base_stream"
+    name: "attachments_compact"
+    schema_loader:
+      type: InlineSchemaLoader
+      schema:
+        $ref: "#/definitions/attachments_compact_schema"
+    $parameters:
+      path: "/attachments"
+    retriever:
+      $ref: "#/definitions/retriever"
+      paginator:
+        $ref: "#/definitions/paginator_base"
+      requester:
+        $ref: "#/definitions/requester"
+        use_cache: true
+        request_parameters:
+          parent: "{{ stream_slice.parent_gid }}"
+      partition_router:
```

Comment:
It is removed because of the duplicate record problem that was commented on [earlier,](https://github.com/airbytehq/airbyte/pull/36951#discussion_r1579545831) should I switch back?

Reply:
Either we could use Tasks or Projects to retrieve the attachments, or we can create a new stream if needed,`attachments_compact_via_tasks`, `attachments_compact_via_Projects`
Is there any better way @askarpets

Reply:
@btkcodedev please add the partition router like below:
```
partition_router:
  - type: SubstreamPartitionRouter
    parent_stream_configs:
      - stream: "#/definitions/projects_stream"
        parent_key: "gid"
        partition_field: "parent_gid"
      - stream: "#/definitions/tasks_stream"
        parent_key: "gid"
        partition_field: "parent_gid"
```

Reply:
Added